Tests/ApolloCodegenTests/CodeGeneration/Templates/OperationDefinitionTemplateTests.swift

```diff
@@ -147,6 +147,31 @@ class OperationDefinitionTemplateTests: XCTestCase {
     expect(actual).to(equalLineByLine(expected, atLine: 3, ignoringExtraLines: true))
   }
 
+  func test__generate__givenQueryWithNameEndingInQuery_generatesQueryOperationWithoutDoubledTypeSuffix() throws {
```

Comment:
👍🏻

Reply:
So if you have a query that someone's incorrectly named the `BlahDeBlahMutation`, would its correctly generated name be `BlahDeBlahMutationQuery`?

Reply:
I don't think that's a bad thing; adhere to the type rather than more-likely-than-not incorrect naming.

Reply:
I wasn't implying that's a bad thing, just validating my assumption. Might be worth adding a test that validates this behavior if it's what we want.

Reply:
Yes, that's correct. I can add another test.